misc.go

```diff
@@ -246,13 +246,16 @@ func postJSON(ctx context.Context, client httpClient, endpoint, token string, js
 }
 
 // post a url encoded form.
-func postForm(ctx context.Context, client httpClient, endpoint string, values url.Values, intf interface{}, d debug) error {
+func postForm(ctx context.Context, client httpClient, endpoint string, values url.Values, intf interface{}, d debug, token ...string) error {
```

Comment:
`postForm` method is not suitable for requests that involve `Content-Type: application/json` headers.
 How about using postJSON from misc.go instead?

Reply:
I must have missed that function! This API method does, in fact, support `application/x-www-form-urlencoded` (as long as you pass the token in the `Authorization` header), but I'll switch it over anyway.

Reply:
Thank you!

( FYI: https://api.slack.com/web#slack-web-api__basics__post-bodies__url-encoded-bodies )

Reply:
I understand that this is Slack's convention, though there's an issue with this particular method: it requires a bearer token in the `Authorizaion` header, regardless of whether or not you're using `application/json`.